api/v1alpha1/search_types.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SearchSpec defines the desired state of Search
+type SearchSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// +optional
+	// Storgeclass Name to be used by default db
+	DBStorage StorageSpec `json:"dbStorage,omitempty"`
+
+	// +optional
+	//Configmap name contains parameters to override default db parameters
+	DbConfig string `json:"dbConfig,omitempty"`
```

Comment:
Does this references a ConfigMap? Referencing another object adds complexity. Why don't we define the required data here? 
Maybe the name could be `ExternalDBInstance`.

Reply:
This is name of the Configmap , isnt it like name of the storageClass @jlpadilla

Reply:
Yes Options 2 - is what we want - But creating a configmap from Key,value pair is easy - But this comes in only when we have to tune the DB.

Reply:
Creating the configmap is easy, but it becomes another step and another object to track, backup, etc.